cmake/WaveUtils.cmake

```diff
@@ -0,0 +1,23 @@
+function(wave_add_test NAME)
```

Comment:
I like this, one nit-pick thing though, I personally would like to have a CMake style rule that if its CMAKE commands it should be all caps? Else we now have a mix between lower and ALL CAPS CMake command calls?